tests/cursor/bug1839-001.phpt

```diff
@@ -0,0 +1,54 @@
+--TEST--
+PHPC-1839: Referenced, out-of-scope, non-interned string in typeMap
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_not_clean(); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+function createTypemap()
+{
+    // Assemble the string so as to not have an interned string
+    $rootValue = chr(ord('a')) . 'rray';
+    $documentValue = chr(ord('a')) . 'rray';
+
+    // Use a reference to this non-interned string in the type map
+    $typemap = ['root' => &$rootValue, 'document' => &$documentValue];
+
+    return $typemap;
+}
+
+$typemap = createTypemap();
+
+$manager = new MongoDB\Driver\Manager(URI);
+$cursor = $manager->executeQuery(NS, new MongoDB\Driver\Query([]));
+
+echo "Before:\n";
+debug_zval_dump($typemap);
+
+$cursor->setTypemap($typemap);
+
+echo "After:\n";
+debug_zval_dump($typemap);
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+Before:
+array(2) refcount(2){
+  ["root"]=>
+  string(5) "array" refcount(1)
```

Comment:
Despite us assigning a reference, this is apparently converted by PHP once the variable goes out of scope. I was able to confirm this on all PHP versions: https://3v4l.org/12fdR. For this reason, I decided to keep both tests to make sure we're not running into strange behaviour anymore; this took enough time to discover already.

Reply:
I discovered that this is indeed intended: `debug_zval_dump` only shows the reference indicator (`&`) for references with a refCount > 1.

Reply:
Can you clarify if the value of "root" (and other keys) here _are_ still references, and this is just a matter of the behavior of https://github.com/php/php-src/commit/2eb980f614ab59aa8f632ae5a70f719f34e8410f deciding not to render them as such?

If so, perhaps note that in a comment on the first call to `debug_zval_dump()` in the test.

----

Based on my reading of [Dereferencing and Unwrapping](https://www.phpinternalsbook.com/php7/zvals/references.html#dereferencing-and-unwrapping), the unwrapping in `php_array_api.h` should only affect the internal zval. I'd expect the reference in the `$typeMap` array to remain as-is.